cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -337,6 +339,41 @@ struct ISOWeek {
   Localizer localizer_;
 };
```

Comment:
Given ISOWeek is a special case of Week now, maybe we should just have one kernel (and perhaps some conveniences, e.g. DayOfWeekOptions::IsoWeek())?

Reply:
Oh yeah, that's a nice idea.
I'm still trying to exactly match `WEEK` from MySQL and once I do I'll look into this. Sorry for the WIP.

Reply:
Oh, no worries. I can hold off looking at this until you're ready.

Reply:
Early review is great though! :)

Reply:
Ping on this? Since it seems the two kernels are very similar.